pyomo/opt/plugins/dakota_text_io.py

```diff
@@ -17,11 +17,27 @@
 """
 
 import re
-
-from pyutilib.math import as_number
+import six
 from pyomo.opt.blackbox.problem_io import BlackBoxOptProblemIOFactory
 
 
+def as_number(value):
+    if type(value) in [int, float]:
+        return value
+    if isinstance(value, six.string_types):
+        try:
+            tmp = int(value)
+            return tmp
+        except ValueError:
+            pass
+        try:
+            tmp = float(value)
+            return tmp
+        except ValueError:
+            pass
+    return value
```

Comment:
Is this copied from pyutilib.math? Is this the best location for this function? I'm guessing you put it here as this is the only spot where it's being used.

Reply:
Exactly that logic. I'm open to putting it somewhere else, but it is only used in this spot.